src/http/fetch.rs

```diff
@@ -61,33 +63,48 @@ pub(crate) fn init(ctx: &Ctx<'_>, globals: &Object) -> Result<()> {
             async move {
                 let options = options?;
 
-                let uri: Uri = options.url.parse().or_throw(&ctx)?;
+                let mut uri: Uri = options.url.parse().or_throw(&ctx)?;
                 let method_string = options.method.to_string();
                 let method = options.method;
                 let abort_receiver = options.abort_receiver;
 
                 ensure_url_access(&ctx, &uri)?;
 
-                let mut req = Request::builder()
-                    .method(method)
-                    .uri(uri)
-                    .header("user-agent", format!("llrt {}", VERSION))
-                    .header("accept", "*/*");
+                let mut redirect_count = 0;
+                let res = loop {
+                    let req = build_request(&ctx, &method, &uri, &options.headers, &options.body)?;
+
+                    let res = if let Some(abort_receiver) = &abort_receiver {
+                        select! {
+                            res = client.request(req) => res.or_throw(&ctx)?,
+                            reason = abort_receiver.recv() => return Err(ctx.throw(reason))
+                        }
+                    } else {
+                        client.request(req).await.or_throw(&ctx)?
+                    };
 
-                if let Some(headers) = options.headers {
-                    for (key, value) in headers.iter() {
-                        req = req.header(key, value)
+                    match res.headers().get(HeaderName::from_static("location")) {
+                        Some(location_headers) => {
+                            if let Ok(location_str) = location_headers.to_str() {
+                                uri = location_str.parse().or_throw(&ctx)?;
+                                ensure_url_access(&ctx, &uri)?;
+                            }
+                        },
+                        None => break res,
+                    };
+
+                    if options.redirect == "manual" {
+                        return Err(Exception::throw_type(&ctx, "Unsupported redirect option"));
```

Comment:
We can just ignore the redirect here and return the raw response

Reply:
We have corrected the problem, please confirm.